hack/verify-go-modules.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script verifies all go modules in the repository.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+
+mapfile -t DIRS < <(find "${REPO_ROOT}" -name go.mod -print0 | xargs -0 dirname)
+
+for dir in "${DIRS[@]}"; do
```

Comment:
This is a lot of logic for not much good.

```
bash-5.1$ find . -name go.mod
./ta
ste/go.mod
./go.mod
./docs/go.mod
./te  st/go.mod
./pkg/apis/go.mod

bash-5.1$ mapfile -t DIRS < <(find . -name go.mod -print0 | xargs -0 dirname)

bash-5.1$ for dir in "${DIRS[@]}"; do echo working on "($dir)"; done
working on (./ta)
working on (ste)
working on (.)
working on (./docs)
working on (./te  st)
working on (./pkg/apis)
```

Reply:
cf

```
bash-5.1$ cat > process.sh
echo "Working on (${1#go.mod})"

bash-5.1$ chmod a+x process.sh

bash-5.1$ find . -name go.mod -exec $PWD/process.sh \{\} \;
Working on (./ta
ste/go.mod)
Working on (./go.mod)
Working on (./docs/go.mod)
Working on (./te  st/go.mod)
Working on (./pkg/apis/go.mod)
```

Reply:
cf

```
bash-5.1$ cat > process.sh
#!/usr/bin/env bash
echo "Working on (${1%/go.mod})"
bash-5.1$ 
bash-5.1$ chmod a+x process.sh
bash-5.1$ find . -name go.mod -exec $PWD/process.sh \{\} \;
Working on (./ta
ste)
Working on (.)
Working on (./docs)
Working on (./te  st)
Working on (./pkg/apis)
```

Reply:
removed file altogether since it is not being used